src/simulation/m_global_parameters.fpp

```diff
@@ -1035,6 +1048,16 @@ contains
             grid_geometry = 3
         end if
 
+        if (chemistry) then
+            chemistry_idx%beg = sys_size + 1
```

Comment:
What if something else takes up this index? By setting `chemistry_idx%beg = sys_size + 1`, you are asserting that chemistry will be the last set variable (ever), so any additional physics cannot be added.

Reply:
I don't see how this is true. This is the pattern:

```fortran
if (feature) then
    feature_idx%beg = sys_size + 1
    feature_idx%end = sys_size + num_feature_equations
    sys_size = feature_idx%end
end if
```

This scales to any number of equations and new features. This is what some parts of the code already do, though not this concisely. We just keep increasing `sys_size` and assigning indices accordingly. Nothing is stopping you from creating variables after chemistry ones, just increment `sys_size` accordingly.